src/main/java/org/simplify4u/plugins/ValidationChecksum.java

```diff
@@ -0,0 +1,136 @@
+        final byte[] checksumPriorValidation;
+        try {
+            log.debug("Looking for prior validation checksum at: " + file.getAbsolutePath());
+            checksumPriorValidation = FileUtils.readFileToByteArray(file);
+        } catch (final FileNotFoundException e) {
+            log.debug("No prior validation executed. Continuing with full artifact validation.");
+            return false;
+        } catch (final IOException e) {
+            log.debug("Validation of artifacts against prior validation run failed.", e);
+            return false;
+        }
+
+        return Arrays.equals(this.checksum, checksumPriorValidation);
+    }
+
+    /**
+     * Save current checksum to file.
+     */
+    void saveChecksum() {
+        try {
+            FileUtils.writeByteArrayToFile(file, checksum);
+        } catch (final IOException e) {
+            log.debug("Failed to save checksum after successful artifact validation.", e);
+        }
+    }
+
+    /**
+     * A builder for the ValidationChecksum.
+     */
+    static final class Builder {
+
+        private static final String FILENAME_CHECKSUM_PRIOR_VALIDATION = "pgpverify-prior-validation-checksum";
+
+        private final Log log;
+        private final File file;
+
+        /**
+         * Constructor for a builder that uses the given logger and performs file system
+         * operations in a given directory.
+         *
+         * @param log the logger instance
+         * @param directory the target directory
+         */
+        Builder(Log log, File directory) {
+            this.log = requireNonNull(log);
+            this.file = new File(directory, FILENAME_CHECKSUM_PRIOR_VALIDATION);
+        }
+
+        /**
+         * Perform checksum calculation on artifacts.
+         *
+         * @param artifacts the artifacts as deterministically ordered collection
+         * @return Returns the validation checksum instance.
+         */
+        ValidationChecksum checksum(Iterable<Artifact> artifacts) {
+            return new ValidationChecksum(this.log, this.file, calculateChecksum(artifacts));
+        }
+
+        private byte[] calculateChecksum(Iterable<Artifact> artifacts) {
+            final SHA256Digest digest = new SHA256Digest();
+            final byte[] result = new byte[digest.getDigestSize()];
+            for (final Artifact artifact : artifacts) {
+                final byte[] id = artifact.getId().getBytes(UTF_8);
+                digest.update(id, 0, id.length);
+                digest.update((byte) '\0');
+            }
+            digest.doFinal(result, 0);
+            log.debug("Checksum of resolved artifacts: " + ByteUtils.toHexString(result, "0x", ""));
+            return result;
+        }
+
+        /**
+         * Delete a checksum if present in the specified directory.
+         *
+         * @param directory the target directory for a checksum file.
+         */
+        static void deleteChecksum(File directory) {
```

Comment:
only used in test, so test should do it in your scope

Reply:
I prefer to put it in the production class, as this allows me to keep the file name in use "hidden" (encapsulated). We can always make the constant package-private but that means the user needs to be aware of the constant.

I do not have a strong preference here, so just let me know if you find my argument acceptable or not.

Reply:
ok, it can stay